Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CentralRepoPostgresSettingsUtil.java

```diff
@@ -0,0 +1,177 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.centralrepository.datamodel;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import org.sleuthkit.autopsy.core.UserPreferences;
+import org.sleuthkit.autopsy.core.UserPreferencesException;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.coreutils.ModuleSettings;
+import org.sleuthkit.autopsy.coreutils.TextConverter;
+import org.sleuthkit.autopsy.coreutils.TextConverterException;
+import org.sleuthkit.datamodel.CaseDbConnectionInfo;
+
+/**
+ * handles saving and loading of postgres settings for central repo
+ */
+public class CentralRepoPostgresSettingsUtil {
+    private final static Logger LOGGER = Logger.getLogger(CentralRepoPostgresSettingsUtil.class.getName());
+    
+    private final static String DEFAULT_HOST = ""; // NON-NLS
+    private final static int DEFAULT_PORT = 5432;
+    private final static String DEFAULT_DBNAME = "central_repository"; // NON-NLS
+    private final static String DEFAULT_USERNAME = "";
+    private final static String DEFAULT_PASSWORD = "";
+
+    private static final String PASSWORD_KEY = "db.postgresql.password";
+    private static final String BULK_THRESHOLD_KEY = "db.postgresql.bulkThreshold";
+    private static final String PORT_KEY = "db.postgresql.port";
+    private static final String USER_KEY = "db.postgresql.user";
+    private static final String DBNAME_KEY = "db.postgresql.dbName";
+    private static final String HOST_KEY = "db.postgresql.host";
+
+    private static final String MODULE_KEY = "CentralRepository";
+
+    
+    
+    private static String valOrDefault(String val, String defaultVal) {
+        if (val == null || val.isEmpty())
+            return defaultVal;
+        
+        return val;
+    }
+    
+    private static int valOrDefault(String val, int defaultVal, Integer min, Integer max) {
+        try {
+            if (val == null || val.isEmpty()) {
+                return defaultVal;
+            } else {
+                int retVal = Integer.parseInt(val);
+                if ((min != null && retVal < min) || (max != null && retVal > max)) {
+                    return defaultVal;
+                }
+                else {
+                    return retVal;
+                }
+            }
+        } catch (NumberFormatException ex) {
+            return defaultVal;
+        }
+    }
+    
+    private static void handleTry(TryHandler handler) {
```

Comment:
This could use a comment and a more descriptive name. As it stands, logException() would be more accurate.